### ISSUE TITLE
local defiads node working with local regtest network

### ISSUE THREAD

**Issue by philbertw4** [defiads/defiads#18]
I've begun trying to get defiads working locally with a local bitcoin core node serving a regtest network. Unfortunately I ran into some snags:

After running `target/release/defiads --bitcoin-network regtest --bitcoin-peers 127.0.0.1:18443` I was greeted with: 
```
19:18:14 [ERROR] thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `regtest`,
 right: `testnet`': src/defiads/main.rs:246
```
This might be due to some of the dependency stuff in #7 we were discussing. I need to try it with Jonas' cargo.lock instead of the one I had in #17. 

Anyway, I'll figure out how to resolve this but just wanted to open the issue so that it's clear what I am currently working on.

My goal is to get defiads running locally with regtest and then add some documentation to the readme for how others can do the same. Hopefully this will help bring us in sync.

**Comment by philbertw4** [comment]
I'm (slowly) making some progress on this. The issue seems to be that, despite the node being started with the `--bitcoin-network regtest` flag, defiads seems to put something in its config file that indicates (incorrectly) `testnet` when it should be `regtest`.

Curiously, on the very first run of defiads, when no `~/.defiads/` directory exists yet, the assertion error does not happen. So, I think this means that somehow one or more of the settings in `~/.defiads/regtest/defiads.cfg` are pointing to `testnet`.

**Comment by philbertw4** [comment]
Sadly I cannot seem to get the assertion on this line to succeed:
https://github.com/defiads/defiads/blob/master/src/defiads/main.rs#L246

More likely than not, this is due to my rust n00bness. I'll keep trying, but this is also something that a rust veteran could likely solve in 30 seconds.

**Comment by notmandatory** [comment]
@philbertw4 I also ran into this issue and did a quick fix in my fork, I'll put together a little MR for it, but here's a preview: https://github.com/notmandatory/defiads/commit/9f68f04bcc58d3ebbae8a38d9fe55402130480b0

**Comment by notmandatory** [comment]
I created PR #22 for this issue

**Comment by philbertw4** [comment]
Great! Thank you! So it appears that the culprit was in fact that both testnet and regtest keys were decoding as being `Network::Testnet` by the underlying non-defiads library. It also explains why commenting out the assertion (which I did while debugging) didn't seem to cause any problems.

**Comment by notmandatory** [comment]
yes, the xpub [bip32](https://github.com/bitcoin/bips/blob/8677fd57864430becd4e828f3ce625922c91b4fd/bip-0032.mediawiki#serialization-format) only seems to have a magic number for mainnet and testnet, so regtest uses testnet.

**Comment by philbertw4** [comment]
fixed by PR #22, closing this issue.
